Multisig/Presentation/UI Library/QRScanner/QRCodeScannerViewController.swift

```diff
@@ -0,0 +1,158 @@
+        } else {
+            presentFaildToCreateScannerAlert()
+            return
+        }
+
+        let metadataOutput = AVCaptureMetadataOutput()
+
+        if (captureSession.canAddOutput(metadataOutput)) {
+            captureSession.addOutput(metadataOutput)
+            metadataOutput.setMetadataObjectsDelegate(self, queue: DispatchQueue.main)
+            metadataOutput.metadataObjectTypes = [.qr]
+        } else {
+            presentFaildToCreateScannerAlert()
+            return
+        }
+
+        previewLayer = AVCaptureVideoPreviewLayer(session: captureSession)
+        previewLayer.frame = view.layer.bounds
+        previewLayer.videoGravity = .resizeAspectFill
+        view.layer.insertSublayer(previewLayer, at: 0)
+
+        captureSession.startRunning()
+    }
+    
+    private func presentCameraAccessRequiredAlert() {
+        let settingsAppURL = URL(string: UIApplication.openSettingsURLString)!
+        let alert = UIAlertController(
+            title: Strings.cameraAlertTitle,
+            message: Strings.cameraAlertMessage,
+            preferredStyle: .alert
+        )
+        alert.addAction(UIAlertAction(title: Strings.cameraAlertAllow, style: .cancel) { _ in
+            UIApplication.shared.open(settingsAppURL, options: [:], completionHandler: nil)
+        })
+        
+        alert.addAction(UIAlertAction(title: Strings.cameraAlertCancel, style: .default) { [weak self] _ in
+            self?.dismiss(animated: true, completion: nil)
+        })
+        present(alert, animated: true)
+    }
+
+    func presentFaildToCreateScannerAlert() {
+        let ac = UIAlertController(title: Strings.scannerNotSupportedTitle, message: Strings.scannerNotSupportedMessage, preferredStyle: .alert)
+        ac.addAction(UIAlertAction(title: Strings.ok, style: .default))
+        present(ac, animated: true)
+        captureSession = nil
+    }
+    
+    func scannerDidScan(code: String) {
+        self.delegate?.scannerViewControllerDidScan(code)
+        captureSession.stopRunning()
+        dismiss(animated: true, completion: nil)
+    }
+
+    override var preferredStatusBarStyle: UIStatusBarStyle {
+        return .lightContent
+    }
+}
+
+extension QRCodeScannerViewController: AVCaptureMetadataOutputObjectsDelegate {
+    func metadataOutput(_ output: AVCaptureMetadataOutput, didOutput metadataObjects: [AVMetadataObject], from connection: AVCaptureConnection) {
+        let metadataObject = metadataObjects.first { (metaObject) -> Bool in
+            guard let readableObject = metaObject as? AVMetadataMachineReadableCodeObject else { return false}
+            return !(readableObject.stringValue?.isEmpty ?? false) && readableObject.type == .qr
+        } as? AVMetadataMachineReadableCodeObject
+        
+        if let metadataObject = metadataObject {
+            AudioServicesPlaySystemSound(SystemSoundID(kSystemSoundID_Vibrate))
+            scannerDidScan(code: metadataObject.stringValue!)
```

Comment:
I think rewriting this method will be more readable:
```suggestion
    func metadataOutput(_ output: AVCaptureMetadataOutput,
                        didOutput metadataObjects: [AVMetadataObject],
                        from connection: AVCaptureConnection) {
        let scannedValue = metadataObjects
            .compactMap { $0 as? AVMetadataMachineReadableCodeObject }
            .filter { $0.type == .qr }
            .compactMap { $0.stringValue }
            .filter { !$0.isEmpty }
            .first
        
        if let code = scannedValue {
            AudioServicesPlaySystemSound(SystemSoundID(kSystemSoundID_Vibrate))
            scannerDidScan(code: code)
        }
    }
```